### PR TITLE
span SetAttribute crash

### LINKED ISSUE #1282: span SetAttribute occur crash
When I execute Span::End, if I execute Span::SetAttribute again, a crash will occur. I checked the implementation and found that Span::SetAttribute does not execute to determine whether the recordable_ is null, but other interfaces have been executed.
image
wecom-temp-459b18201251f1904171a86c44fe64e5

### PR DESCRIPTION
**Pull request by esigo**
Fixes #1282 (issue)

## Changes

adds null check and unit test
For significant contributions please make sure you have completed the following items:

* [x] `CHANGELOG.md` updated for non-trivial changes
* [x] Unit tests have been added
* [ ] Changes in public API reviewed

### CODE REVIEW

File: sdk/src/trace/span.cc [hunk 1]
```diff
@@ -94,7 +94,10 @@ Span::~Span()
 void Span::SetAttribute(nostd::string_view key, const common::AttributeValue &value) noexcept
 {
   std::lock_guard<std::mutex> lock_guard{mu_};
-
+  if (recordable_ == nullptr)
+  {
+    return;
+  }
```
Comment (reyang) [inline sdk/src/trace/span.cc:100]: nit, consider adding an empty line between L100 and L101

**Review by reyang — APPROVED** [review]
LGTM, perhaps worth a changelog entry?

File: sdk/src/trace/span.cc (hunk 1 above)
Comment (esigo) [reply sdk/src/trace/span.cc:100]: thanks, done

File: sdk/src/trace/span.cc [hunk 2]
```diff
@@ -94,6 +94,10 @@ Span::~Span()
 void Span::SetAttribute(nostd::string_view key, const common::AttributeValue &value) noexcept
 {
   std::lock_guard<std::mutex> lock_guard{mu_};
+  if (recordable_ == nullptr)
+  {
```
Comment (ThomsonTan) [inline sdk/src/trace/span.cc:98]: Do we need emit a log message in this case (`recordable_ == nullptr`)?

**Review by ThomsonTan — APPROVED** [review]

File: sdk/src/trace/span.cc (hunk 2 above)
Comment (esigo) [reply sdk/src/trace/span.cc:98]: We don't emit in the other methods.

**Review by owent — APPROVED** [review]
LGTM. Should we also add unit tests to call `AddEvent` , `SetStatus` and `UpdateName` after `End`?
They have simular problems.

**Comment by esigo** [comment]
> LGTM. Should we also add unit tests to call `AddEvent` , `SetStatus` and `UpdateName` after `End`? They have simular problems.

Thanks, added more tests after `End` to cover other methods.

**Pull request merged**